src/browser/client-scripts/rect.js

```diff
@@ -164,3 +164,76 @@ exports.getAbsoluteClientRect = function getAbsoluteClientRect(element, opts) {
 
     return clientRect.translate(util.getScrollLeft(opts.scrollElem), util.getScrollTop(opts.scrollElem));
 };
+
+function getNestedBoundingClientRect(node, boundaryWindow) {
```

Comment:
Found solution in this package - https://github.com/suchipi/get-nested-bounding-client-rect/tree/master. The basic logic is taken from it with some modification for our use case.